golang/Makefile

```diff
@@ -0,0 +1,58 @@
+GODEPS = $(realpath ./Godeps/_workspace)
+GOPATH := $(GODEPS):$(GOPATH)
+PATH := $(GODEPS)/bin:$(PATH)
+PKGS := ./  ./typed
+BUILD := ./build
+
+# target: all - Run tests
+all: test
+
+# target: help - Display targets
+help:
+	@egrep "^# target:" [Mm]akefile | sort -
+
+# target: clean - Cleans build artifacts
+clean:
+	echo Cleaning build artifacts...
+	go clean
+	rm -rf $(BUILD)
+	echo
+
+# target: fmt - Formats go code
+fmt format:
+	echo Formatting Packages...
+	go fmt $(PKGS) 
+	echo
+
+# target: test - Runs tests
+test: clean 
+	echo Testing packages:
+	mkdir -p $(BUILD)
+	go test $(PKGS) $(TEST_ARG) -parallel=4
+
+# target: cover - Runs tests under code coverage
+cover: clean
+	echo Testing packages:
+	mkdir -p $(BUILD)
+	go test ./ $(TEST_ARG)  -coverprofile=$(BUILD)/coverage.out
+	go tool cover -html=$(BUILD)/coverage.out
+	# TODO(mmihic): Temporarily disabled while working out false positives
```

Comment:
yes annoyingly go-logging results in false positives from vet due to having an Error method.  There's a TODO to remove the go-logging dependency which will resolve; I held off doing that in this branch but can do so.